pkg/fieldpath/paved_test.go

```diff
@@ -161,6 +161,14 @@ func TestGetValue(t *testing.T) {
 				err: errors.Wrap(errors.New("unexpected ']' at position 5"), "cannot parse path \"spec[]\""),
 			},
 		},
+		"NilValue": {
+			reason: "Requesting for an object that has nil value",
```

Comment:
nit: I think we had better tell the nil value is not a leaf but rather an intermediate component in the requested path (a parent):
```suggestion
		"NilParent": {
			reason: "Request for a path with a nil parent value",
```

Reply:
Done!